File: Golang/Medium/2048_Next_Greater_Numerically_Balanced_Number/solution.go
package nextgreaternumericallybalancednumber

func nextBeautifulNumber(n int) int {
	n += 1
	for !isNumBalance(n) {
		n += 1
	}
	return n
}

func isNumBalance(n int) bool {
	numMap := make(map[int]int)

	for n > 0 {
		r := n % 10
		numMap[r] += 1
		n = n / 10
	}

	for num, value := range numMap {
		if num != value {
			return false
		}
	}
	return true
}

File: Python/Medium/2048_Next_Greater_Numerically_Balanced_Number/solution.py
class Solution:
    def nextBeautifulNumber(self, n: int) -> int:
        n+=1
        while not Solution.isNumBalance(n):
            n += 1
        return n

    def isNumBalance(n):
        numMap = {}
        while n > 0:
            r = n % 10
            if r not in numMap:
                numMap[r] = 0
            numMap[r] += 1
            n = n//10

        for num in numMap:
            if numMap[num] != num:
                return False
        return True